bin/check-http.rb

```diff
@@ -28,6 +28,9 @@
 #   Use a proxy to check a URL
 #   check-http.rb -u https://www.google.com --proxy-url http://my.proxy.com:3128
 #
+#   Use a proxy with username and password to check a URL
+#   check-http.rb -u https://www.google.com --proxy-url http://a_user:a_pass@my.proxy.com:3128
```

Comment:
I think we should call out the danger of specifying credentials via CLI args and encourage people to use token substitution to avoid it being exposed on the process table, logs, alerts, etc.

Reply:
Good point! I added a check for environment variables PROXY_USER and PROXY_PASS.